demo/app/src/App.js

```diff
@@ -0,0 +1,349 @@
+import React, { useState, useEffect } from "react";
```

Comment:
A couple of notes on this file:
1. Since this component needs to maintain state, especially on this scale, i would expect it to be a class instead of functional.
2. Love that you are using components as children, but this file should probably be refactored to be more readable and maintainable. For example, App.js could maybe have just have the `<StatefulTabs />` Component . Then that component could handle the logic for what dom elements to show based on its own complex logic.
Then maybe the checkboxes could be an iterated array of plugins/features/etc and know how assign all of its attributes on its return `<Checkbox />.

Reply:
Per number 1: I just looked up the hook apis you used and your way is all totally valid. I think its a great solution but is probably best suited for the smaller functional components.

Reply:
1. agreed.
2. the plan is to refactor it and break it down to smaller components as you mentioned. I have a new UI proposal and once we agree on the next steps I will work on refactoring. Just wanted to roll this out "as is" and to be functional so others can give input.
3. agreed: the main layout structure could be improved, but I just took it off from the original demo index.html